File: backend/src/main/java/com/calonuria/backend/dto/user/UsuarioRespuestaDTO.java
package com.calonuria.backend.dto.user;

import lombok.Data;
import lombok.AllArgsConstructor;

@Data
@AllArgsConstructor
public class UsuarioRespuestaDTO {
    private Long idUsuario;
    private String nombre;
    private String email;
    private String rol;
}

File: frontend/between_pages/lib/models/user/usuario_respuesta_dto.dart
class UsuarioRespuestaDTO {
  final int idUsuario;
  final String nombre;
  final String email;
  final String rol;

  UsuarioRespuestaDTO({
    required this.idUsuario,
    required this.nombre,
    required this.email,
    required this.rol,
  });

  factory UsuarioRespuestaDTO.fromJson(Map<String, dynamic> json) {
    return UsuarioRespuestaDTO(
      idUsuario: json['idUsuario'] as int,
      nombre: json['nombre'] as String,
      email: json['email'] as String,
      rol: json['rol'] as String,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'idUsuario': idUsuario,
      'nombre': nombre,
      'email': email,
      'rol': rol,
    };
  }
}
